disnake/state.py

```diff
@@ -1292,24 +1298,30 @@ def parse_guild_member_update(self, data: gateway.GuildMemberUpdateEvent) -> Non
         if member is not None:
             old_member = Member._copy(member)
             member._update(data)
-            user_update = member._update_inner_user(user)
+            user_update = member._update_inner_user(data["user"])
             if user_update:
                 self.dispatch("user_update", user_update[0], user_update[1])
 
             self.dispatch("member_update", old_member, member)
+            user = member
         else:
```

Comment:
To resolve the `TODO` here, the code could be simplified to this:
```py
        else:
            member = Member(data=data, guild=guild, state=self)

            # Force an update on the inner user if necessary
            user_update = member._update_inner_user(data["user"])
            if user_update:
                self.dispatch("user_update", user_update[0], user_update[1])

            if self.member_cache_flags.joined:
                guild._add_member(member)
```

Since the code would then always produce a full `Member` object, `RawGuildMemberUpdateEvent` can probably be removed, unless I'm missing something here.
(and `raw_member_update` would just receive the `Member` object)

cc @onerandomusername since you commented on the previous PR's conversation about this as well

Reply:
Yeah, that would make sense